### PR TITLE
C++11 : Use using instead of typedef

### PR DESCRIPTION
**Pull request by ya1gaurav**
C++11 type alias.

### CODE REVIEW

File: src/lib_json/json_reader.cpp [hunk 1]
```diff
@@ -55,9 +55,9 @@ static size_t const stackLimit_g = JSONCPP_DEPRECATED_STACK_LIMIT; // see readVa
 namespace Json {
 
 #if __cplusplus >= 201103L || (defined(_CPPLIB_VER) && _CPPLIB_VER >= 520)
-typedef std::unique_ptr<CharReader> CharReaderPtr;
+using CharReaderPtr =  std::unique_ptr<CharReader>;
```
Comment (yiqiju) [inline src/lib_json/json_reader.cpp]: I am sorry, the extra space before "std..." hurt my eyes...

**Comment by ya1gaurav** [comment]
@yiqiju Thanks, fixed.
@All members, please review. If it seems fine, I will extend changes to header files also.

**Comment by cdunn2001** [comment]
I have no strong objection, but could you provide some justification for this change?

**Comment by ya1gaurav** [comment]
It is just coding style according to new c++ standards.
http://en.cppreference.com/w/cpp/language/type_alias

**Comment by hjmjohnson** [comment]
these style changes have already been incorporated upstream.

**Pull request closed without merging**